packages/backend-openapi-utils/config.d.ts

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+export interface Config {
+  openapi?: {
+    /**
+     * A list of plugins, whose OpenAPI specs you want to collate in `InternalOpenApiDocumentationProvider`.
+     */
+    plugins: string[];
```

Comment:
This kinda makes it sound like we've got plugins for OpenAPI imo. I think it would be best if the provider follows the same config pattern as other provider, e.g. https://github.com/backstage/backstage/blob/8960902a4ea002d0a44c376f071048b0ef90053f/plugins/catalog-backend-module-github/src/providers/GithubEntityProviderConfig.ts#L50 https://github.com/backstage/backstage/blob/fa7004d9722c7f8290acaa5842af769512b7cf3e/plugins/catalog-backend-module-github/config.d.ts#L57

The equivalent of the example config in this PR would be

```yaml
catalog:
  providers:
    openapi:
      plugins:
        - catalog
        - search
        - todo
```

Jumping ahead a bit this could also allow this to be a more generic OpenAPI provider, or if that's a bad idea we could name it something more specific plugin `catalog.providers.pluginOpenapiSpec`